src/sql/workbench/parts/notebook/cellViews/stdin.component.ts

```diff
@@ -0,0 +1,120 @@
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { inputBackground, inputBorder } from 'vs/platform/theme/common/colorRegistry';
+import { StandardKeyboardEvent, IKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { KeyCode } from 'vs/base/common/keyCodes';
+import { AngularDisposable } from 'sql/base/node/lifecycle';
+import * as DOM from 'vs/base/browser/dom';
+import { Deferred } from 'sql/base/common/promise';
+import { ICellModel, CellExecutionState } from 'sql/workbench/parts/notebook/models/modelInterfaces';
+
+export const STDIN_SELECTOR: string = 'stdin-component';
+@Component({
+	selector: STDIN_SELECTOR,
+	template: `
+		<div class="prompt">{{prompt}}</div>
+		<div #input class="input"></div>
+	`
+})
+export class StdInComponent extends AngularDisposable implements AfterViewInit {
+	private _input: InputBox;
+	@ViewChild('input', { read: ElementRef }) private _inputContainer: ElementRef;
+
+	@Input() stdIn: nb.IStdinMessage;
+	@Input() onSendInput: Deferred<string>;
+	@Input() cellModel: ICellModel;
+
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private changeRef: ChangeDetectorRef,
+		@Inject(IWorkbenchThemeService) private themeService: IWorkbenchThemeService,
+		@Inject(IContextViewService) private contextViewService: IContextViewService,
+		@Inject(forwardRef(() => ElementRef)) private el: ElementRef
+	) {
+		super();
+	}
+
+	ngOnInit(): void {
+
+	}
+
+	ngAfterViewInit(): void {
+		let inputOptions: IInputOptions = {
+			placeholder: '',
+			ariaLabel: this.prompt
+		};
+		this._input = new InputBox(this._inputContainer.nativeElement, this.contextViewService, inputOptions);
+		if (this.password) {
+			this._input.inputElement.type = 'password';
+		}
+		this._register(this._input);
+		this._register(attachInputBoxStyler(this._input, this.themeService, {
+			inputValidationInfoBackground: inputBackground,
+			inputValidationInfoBorder: inputBorder,
+		}));
+		if (this.cellModel) {
+			this._register(this.cellModel.onExecutionStateChange((status) => this.handleExecutionChange(status)));
+		}
+		this.onkeydown(this._input.inputElement, (e) => this.handleKeyboardInput(e));
+		this._input.focus();
+	}
+	private handleKeyboardInput(e: IKeyboardEvent): void {
+		switch (e.keyCode) {
+			case KeyCode.Enter:
+				// Indi
+				if (this.onSendInput) {
+					this.onSendInput.resolve(this._input.value);
+				}
+				e.stopPropagation();
+				break;
+			case KeyCode.Escape:
+				if (this.onSendInput) {
+					this.onSendInput.reject('');
+				}
+				e.stopPropagation();
+				break;
+			default:
+				// No-op
+				break;
+		}
+	}
+
+	handleExecutionChange(status: CellExecutionState): void {
+		if (status !== CellExecutionState.Running && this.onSendInput) {
+			this.onSendInput.reject('');
+		}
+	}
+
+
+	private onkeydown(domNode: HTMLElement, listener: (e: IKeyboardEvent) => void): void {
```

Comment:
I feel like this could also be accomplished by doing a @HostListener that listens for keypresses. At the end of the day it's the same thing so no biggie.

Reply:
So I was taking this from JupyterLab, which I guess doesn't have that. I'm open to doing that just not used to the patterns.

Reply:
Nice - it was easy to apply and simple. Thank you!